runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/jobsubmission/InMemoryJobService.java

```diff
@@ -60,17 +60,28 @@
 
   public static InMemoryJobService create(
       Endpoints.ApiServiceDescriptor stagingServiceDescriptor, JobInvoker invoker) {
-    return new InMemoryJobService(stagingServiceDescriptor, invoker);
+    return new InMemoryJobService(stagingServiceDescriptor, (String session) -> "token", invoker);
+  }
+
+  public static InMemoryJobService create(
+      Endpoints.ApiServiceDescriptor stagingServiceDescriptor,
+      Function<String, String> stagingServiceTokenProvider,
```

Comment:
maybe document that this function expects a preparation ID as input.

Reply:
Good point.